server/src/main/java/io/druid/server/coordination/BatchDataSegmentAnnouncer.java

```diff
@@ -95,18 +103,37 @@ public DataSegment apply(DataSegment input)
         return rv;
       }
     };
+
+    if (this.config.isSkipSegmentAnnouncementOnZk()) {
```

Comment:
history keeper needs access to list of all segments served by the node to support first segment list call (after that deltas are delivered). this list of segments is already maintained in-memory inside BatchDataSegmentAnnouncer and I did not want to clone that list in another place.
Removed `getSegmentChangesSince(..)` from `DataSegmentAnnouncer` interface and made `SegmentListerResource` directly use `BatchDataSegmentAnnouncer`